k8s.io/configmap-nginx.yaml

```diff
@@ -82,7 +82,7 @@ data:
         server_name sigs.k8s.io;
         listen 443 ssl;
 
-        location ~ ^/(?<sig_repo>[^/]*)(?<repo_subpath>/.*)?$ {
+        location ~ ^/(?<sig_repo>.*?)(?!/+$)(?<repo_subpath>/.*)?$ {
```

Comment:
This is a complicated enough regex that I don't want it.  Can you pklease explain to me what it is doing and why it is needed?

Reply:
Sure, I was also hesitant first to touch it.

Reply:
I don't know what `(?!/+$) ` represents.

Can you show me a match that was wrong before and is right after this change?

Reply:
Sure, thats the current implementation:
https://regex101.com/r/jRLEd8/1/

You see that the first match, matches the group `repo_subpath` to `/`, which fails the redirect.

The change I propose:
https://regex101.com/r/8LFNx3/1

This group `repo_subpath` doesn't match at all, only if there is anything behind it.

`(?!/+$)` drops any '/' if there is nothing else until the end of line.

Reply:
I feel like a dunce,.  Thank you for trying to educate me.  I see now that the old form forces the / to be in group which would be wrong considering how it is used.

A more obvious (to me, not a regex wizard) might be: `^\/(?<sig_repo>[^\/]*)(\/(?<repo_subpath>.*)?)$`

This sets up a middle capture group (unnamed) for the / between repo and path, changing the usage below to require adding a / (which actually makes it more obvious to my eyes). 

We could make it a non-capture group, but I don't know that it is warranted - less is more?

You know what?  After staring at yours, I get it now.  OK.  Can you just add a comment like:

```
# The ?! block is negative-lookahead to prevent `/repo/` from grouping into (`repo`, `/`) while `/repo/path` will still group as (`repo`, `/path`).
```

One day I will have to come back and look at this :)

Reply:
Perfect, sounds good to me since it's not so obvious.
I added the comment, guess thats a good first approach until we maybe come up with a better solution.